airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/base/IntegrationRunner.java

```diff
@@ -272,8 +273,15 @@ private void readConcurrent(final JsonNode config, final ConfiguredAirbyteCatalo
   }
 
   private void readSerial(final JsonNode config, final ConfiguredAirbyteCatalog catalog, final Optional<JsonNode> stateOptional) throws Exception {
-    try (final AutoCloseableIterator<AirbyteMessage> messageIterator = source.read(config, catalog, stateOptional.orElse(null))) {
+    final AutoCloseableIterator<AirbyteMessage> messageIterator = source.read(config, catalog, stateOptional.orElse(null));
+    try {
       produceMessages(messageIterator, outputRecordCollector);
+      try {
+        messageIterator.close();
+      } catch (Exception e) {
```

Comment:
If we're catching all the exceptions here, why is it required to add the `closeSafely()` method? Which from what I can tell is doing the same thing?

Reply:
apologies, this was a WIP. Let me remove it.